core/src/main/java/org/elasticsearch/action/admin/indices/analyze/TransportAnalyzeAction.java

```diff
@@ -243,11 +272,240 @@ protected AnalyzeResponse shardOperation(AnalyzeRequest request, ShardId shardId
+                Reader reader = new FastStringReader(text);
+                if (charfilterFactories != null) {
+                    for (CharFilterFactory charfilter : charfilterFactories) {
+                        reader = charfilter.create(reader);
+                        Reader readerForWriteOut = new FastStringReader(charFilteredSource);
+                        readerForWriteOut = charfilter.create(readerForWriteOut);
+                        charFilteredSource = writeCharStream(readerForWriteOut);
+
+                        List<String> texts = charFiltersTexts.get(charfilter.name());
+                        if (texts == null) {
+                            texts = Lists.newArrayList();
+                        }
+                        texts.add(charFilteredSource);
+                        charFiltersTexts.put(charfilter.name(), texts);
+                    }
+                }
+
+                //analyzing only tokenizer
+                Tokenizer tokenizer = tokenizerFactory.create();
+                try {
+                    tokenizer.setReader(reader);
+                } catch (IOException e) {
+                    throw new ElasticsearchException("failed to read texts", e);
+                }
+                tokenizerTokenListCreator.analyze(tokenizer, customAnalyzer, field ,includeAttributes, request.shortAttributeName());
+
+                //analyzing each tokenfilter
+                if (tokenfilterFactories != null) {
+                    for (int i = 0; i < tokenfilterFactories.length; i++) {
+                        TokenListCreator tokenfilterTokenListCreator = tokenFiltersTokenListCreator.get(tokenfilterFactories[i].name());
+                        if (tokenfilterTokenListCreator == null) {
+                            tokenfilterTokenListCreator = new TokenListCreator();
+                            tokenFiltersTokenListCreator.put(tokenfilterFactories[i].name(), tokenfilterTokenListCreator);
+                        }
+                        TokenStream stream = createStackedTokenStream(text,
+                            charfilterFactories, tokenizerFactory, tokenfilterFactories, i + 1);
+                        tokenfilterTokenListCreator.analyze(stream, customAnalyzer, field, includeAttributes, request.shortAttributeName());
+                    }
+                }
+            }
+
+            DetailAnalyzeResponse.CharFilteredText[] charfilteredLists = new DetailAnalyzeResponse.CharFilteredText[charFiltersTexts.size()];
+            int i = 0;
+            for (String charFilterName : charFiltersTexts.keySet()) {
+                charfilteredLists[i] = new DetailAnalyzeResponse.CharFilteredText(
+                    charFilterName, charFiltersTexts.get(charFilterName).toArray(Strings.EMPTY_ARRAY));
+                i++;
+            }
+            DetailAnalyzeResponse.AnalyzeTokenList[] tokenfilterLists = new DetailAnalyzeResponse.AnalyzeTokenList[tokenFiltersTokenListCreator.size()];
+            i = 0;
```

Comment:
Good point! I found the problem about the order of tokenfilter in response.
I will fix this problem and I will change implementation.